types/jspdf/index.d.ts

```diff
@@ -76,7 +77,7 @@ declare module 'jspdf' {
         setLineCap(style:string|number):jsPDF;
         setLineJoin(style:string|number):jsPDF;
         output(type?:string, options?:any):any;
-        save(filename:string):jsPDF;
+        save(filename:string, options?: {returnPromise: boolean}): Promise<any>;
```

Comment:
This may cause problem for consumers that rely on the old return type.

Consider use function overload?

```suggestion
        save(filename:string):jsPDF;
        save(filename:string, options: {returnPromise: boolean}): Promise<any>;
```

Reply:
The documentation is wrong.
Looking at the plain vanilla javascript code in the library the save function never return an instance of jsPDF.
Here is the link with the code: https://rawgit.com/MrRio/jsPDF/master/docs/jspdf.js.html#line4706
Inside the if statement "if (options.returnPromise === false)" the code doesn't return any value.
So the returned value is either undefined or a promise.
Also, I am returning a just promise instead of "Promise<any> | undefined" so that when the user invoke save with the returnPromise set to true he can do a concatenation like this:
doc.save('myFile.pdf', {returnPromise: true}).then(()=>{ ... });
instead of checking if it is undefined.

I hope it clarifies my choices.

Thanks

Reply:
I see... Maybe this then?

```suggestion
        save(filename:string): void;
        save(filename:string, options: {returnPromise: boolean}): Promise<any>;
```

Reply:
Ok, that's fine.
I have applied your suggestion.

Thanks!